crwatcher/status.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 the lostromos Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crwatcher
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/mitchellh/mapstructure"
+)
+
+type ResourcePhase string
+
+const (
+	PhaseNone     ResourcePhase = ""
+	PhaseCreating               = "Creating"
+	PhaseRunning                = "Running"
+	PhaseFailed                 = "Failed"
+)
+
+type ConditionReason string
+
+const (
+	// TODO(alecmerdler): Define reasons
```

Comment:
@ecordell @ericavonb @jzelinskie @josephschorr Need some guidance here: what are we expecting for phases/reasons? They should be agnostic about the controller used (Helm/Go templates); the `Message` can contain implementation details.
It doesn't look like Lostromos watches the resources it creates, so we can't know if one of them failed individually.